staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -215,16 +219,30 @@ func createSerializers(config ContentConfig) (*Serializers, error) {
 //  SelectorParam("labels", "area=staging").
 //  Timeout(10*time.Second).
 //  Do()
+// if resp.Status == http.StatusUnauthorized {
+//   success, err := s.Refresh()
+//   if success{
+//   newToken, err := s.Token()
+//   //retry request with new token
+// }
 // if err != nil { ... }
 // list, ok := resp.(*api.PodList)
 //
 func (c *RESTClient) Verb(verb string) *Request {
 	backoff := c.createBackoffMgr()
-
+	var request *Request
 	if c.Client == nil {
-		return NewRequest(nil, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle)
+		request = NewRequest(nil, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle)
+	}
+	request = NewRequest(c.Client, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle)
+	if c.BearerTokenSource != nil {
+		token, err := c.BearerTokenSource.Token()
+		if err != nil {
+			return request
+		}
+		request.SetHeader("Authorization", "Bearer "+token)
```

Comment:
@ericchiang The header from here will override the one from the bearer tokens in the kubeconfig as well as the one from --token cmd line, [code] here(https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/round_trippers.go#L278), I will update here to align with the current logic, and add comments here.